bin/sync/article_updater.rb

```diff
@@ -0,0 +1,162 @@
+require_relative "images_processor.rb"
+require 'yaml'
+require 'json'
+
+class ArticleUpdater
+  JT_BLOG_HOST = 'https://jetthoughts.com/blog/'.freeze
+  YAML_STATUS_FILE = 'sync_status.yml'.freeze
+
+  def initialize(http_client: Net::HTTP, file_manager: File, yaml_parser: YAML, json_parser: JSON)
+    @http_client = http_client
+    @file_manager = file_manager
+    @yaml_parser = yaml_parser
+    @json_parser = json_parser
+  end
+
+  def call(force)
+    articles = force ? all_articles : unsynced_articles
+
+    articles.each do |article_id, attributes|
+      article_data = fetch_article(article_id)
+
+      save_article_as_markdown(article_data, attributes[:slug])
+      download_images(attributes[:slug])
+      update_canonical_url_on_dev_to(article_id, attributes[:slug])
+      update_article_hash(article_id)
+    end
+  end
+
+  private
+
+  def download_images(slug)
+    ImagesProcessor.new(slug).call
```

Comment:
it's a little strange method call download images, but class it called ImagesProcessor, they are not the same ;)

Reply:
`ImageDownloader`?